code/modules/cm_marines/compound_silo.dm

```diff
@@ -0,0 +1,193 @@
+/obj/structure/compound_silo
+	name = "compound silo"
+	desc = "A large silo containing a compound used to speed up dropship engine cooldowns."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "weldtank_old"
+	density = TRUE
+	anchored = TRUE
+	health = 250
+
+	var/compound_type = COMPOUND_PHI
+
+	var/obj/item/nozzle/attached_to
+	var/atom/tether_holder
+	var/tether_range = 12
+
+/obj/structure/compound_silo/omega
+	compound_type = COMPOUND_OMEGA
+
+/obj/structure/compound_silo/tau
+	compound_type = COMPOUND_TAU
+
+/obj/structure/compound_silo/epsilon
+	compound_type = COMPOUND_EPSILON
+
+/obj/structure/compound_silo/Initialize()
+	. = ..()
+
+	name = "[compound_type] silo"
+	attached_to = new /obj/item/nozzle(src, compound_type)
+	RegisterSignal(attached_to, COMSIG_PARENT_PREQDELETED, .proc/override_delete)
+
+/obj/structure/compound_silo/proc/override_delete()
+	SIGNAL_HANDLER
+	recall_nozzle()
+	return COMPONENT_ABORT_QDEL
```

Comment:
if you're gonna use the atrocity that `COMPONENT_ABORT_QDEL` is (is there really a good reason?), please at least check for `force` argument so it still works with things like nightmare or shuttles that use maploading